torch/utils/data/sampler.py

```diff
@@ -112,15 +113,19 @@ def num_samples(self) -> int:
     def __iter__(self) -> Iterator[int]:
         n = len(self.data_source)
         if self.generator is None:
-            self.generator = torch.Generator()
-            self.generator.manual_seed(int(torch.empty((), dtype=torch.int64).random_().item()))
+            if self._gen is None:
```

Comment:
Wait, can you please clarify what is going on here, it looks like you want individual instance of _gen for every iterator, but placing it into `self` not going to help.